File: Semestru 4/Medii de proiectare si programare/Laboratoare/Basket/Basket/models/Ticket.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Baschet.models
{
    public class Ticket : Entity
    {
        public string  idMatch { get; set; }

        public string name { get; set; }
        public int seats { get; set; }

        public Ticket(string idTicket, string idMatch, string name, int seats) {
            this.id = idTicket;
            this.idMatch = idMatch;
            this.name = name;
            this.seats = seats;
        }
        
        public Ticket(string idMatch, string name, int seats) {
            this.idMatch = idMatch;
            this.name = name;
            this.seats = seats;
        }

        
    }
}

File: Semestru 4/Medii de proiectare si programare/Examen/Proiecte/Red/BaschetModel/src/main/java/models/Ticket.java
package models;

import java.util.Objects;

public class Ticket extends Entity{
    String idMatch;
    String name;
    Integer seats;

    public Ticket() {

    }

    public Ticket(String idTicket, String idMatch, String name, Integer seats) {
        this.id = idTicket;
        this.idMatch = idMatch;
        this.name = name;
        this.seats = seats;
    }

    public Ticket(String idMatch, String name, Integer seats) {
        this.idMatch = idMatch;
        this.name = name;
        this.seats = seats;
    }

    public String getIdMatch() {
        return idMatch;
    }

    public void setIdMatch(String idMatch) {
        this.idMatch = idMatch;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getSeats() {
        return seats;
    }

    public void setSeats(Integer seats) {
        this.seats = seats;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Ticket ticket = (Ticket) o;
        return id.equals(ticket.id) && idMatch.equals(ticket.idMatch) && Objects.equals(name, ticket.name) && Objects.equals(seats, ticket.seats);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, idMatch, name, seats);
    }

    @Override
    public String toString() {
        return "Ticket{" +
                "id='" + id + '\'' +
                ", idMatch='" + idMatch + '\'' +
                ", name='" + name + '\'' +
                ", seats='" + seats + '\'' +
                '}';
    }
}
